### PR TITLE
vbatts/backend update

### PR DESCRIPTION
**Pull request by vbatts**
# linters and golang updates

fix inconsistencies in reported linting, and differences in the versions of golang used.

I'm testing this locally with `go1.19.7` now, since the `backend/go.mod` says go1.19.
Though the github workflow was using go1.17 currently, but now is also go1.19.

This also aligns an issue I found where I saw errors by running `make` (i.e. `make all`) locally, but did not see them in `make ci` locally or on the github action. Now that is cleaned up.

## Testing done

```shell
cd backend
make all ci
```
and now the github actions output

Closes: #638

### CODE REVIEW

**Comment by vbatts** [comment]
ok. Ready for review.

File: .github/workflows/backend.yml [hunk 1]
```diff
@@ -31,5 +31,5 @@ jobs:
 
     - name: Build, Lint & Check
       run: |
-        make ci
+        make all ci
```
Comment (yolossn) [inline .github/workflows/backend.yml:34]: The `all` recipe is for building and running both `frontend` and `backend`. Just `make ci` should do here.

**Review by yolossn — CHANGES_REQUESTED** [review]

File: .github/workflows/backend.yml (hunk 1 above)
Comment (vbatts) [reply .github/workflows/backend.yml:34]: ah, this should `cd backend` to run the backend test. The reason I added this at all, is that I was seeing failures locally, that were not getting caught in the github action.

Comment (vbatts) [reply .github/workflows/backend.yml:34]: oh, it is already in the backend directory: `working-directory: backend`
This is not running the frontend build.

Comment (yolossn) [reply .github/workflows/backend.yml:34]: We have a separate action for the frontend.

Comment (vbatts) [reply .github/workflows/backend.yml:34]: correct. This is not running any of the frontend make targets. It's running `make all ci` from within the `./backend/` directory

**Pull request merged**